### DISCUSSION TITLE
A Select Text Field component

### DISCUSSION THREAD

**Discussion by ghost** [mantinedev/mantine#3500] [Feature requests]
## Context
As the name suggests a component that concatenates the values of a select field and a text input into one value using a separator. It would be helpful for normalizing data with prefixes or suffixes like:
- Prefixes for names (Pr. John Doe)
- Currencies ($100)
- Phone numbers (+212-689-456-1234)
- Measure units (10 cm)

## Some props
- `data`: an array of objects containing labels and values for the prefixes/suffixes
- `inputValue`: the value of the text field
- `separator`: by default null or a string, this will be used the concatenate the values
- `value`: the concatenated value
- `left/right`: for the position of the select options

**Comment by rtivital** [comment]
It is not planned to introduce such component. You can use one of the input masking libraries to achieve this - https://mantine.dev/core/input/#inputbase-component

**Reply by ghost** [reply]
Thank you very much, how can I allow users to type `Pr. John Doe`, `Dr. John Doe`, but not `pr.` or `prof.` or `professor`  it would be easier with a component like this
![ScreenShot_20230205100709](https://user-images.githubusercontent.com/88904575/216810712-5a313f7c-84bd-40f4-8a58-d8de51039efa.png)

**Reply by rtivital** [reply]
https://ui.mantine.dev/category/inputs#currency-input

**Reply by ghost** [reply]
Yeah this is a nice one, but does it give the value in the `TextInput` alone or both `TextInput` and the `rightSection`, if not and if a hack with refs is possible just turn it into a reusable component and a lot of users would be happy 👍

**Reply by rtivital** [reply]
> a lot of users would be happy

I doubt that, out of ~2000 feature request, it is the first one that mentions such UI. It is a niche component specific to the application that you are building. It should not be added to the library.

**Reply by ghost** [reply]
> It is a niche component specific to the application that you are building

No, I'm not. I just like this library and I wanted it to have such a component because it helps with the normalization of data input without using masks or regex and reduces the need for client-side validation, and the picture I showed you is from the bootstrap docs it has nothing to do with me.
